imagej/imagej-ops2/src/main/java/net/imagej/ops2/coloc/pValue/DefaultPValue.java

```diff
@@ -89,46 +89,38 @@ public void compute(final RandomAccessibleInterval<T> image1, final RandomAccess
 		final double value = op.apply(trimmedImage1, trimmedImage2);
 
 		// compute shuffled coloc values in parallel
-		int threadCount = Runtime.getRuntime().availableProcessors(); // FIXME: conform to Ops threading strategy...
+		var executor = Parallelization.getTaskExecutor();
+		var numTasks = executor.suggestNumberOfTasks();
+
 		Random r = new Random(seed);
 		long[] seeds = new long[nrRandomizations];
 		for (int s = 0; s < nrRandomizations; s++) {
 			seeds[s] = r.nextLong();
 		}
-		ArrayList<Future<?>> future = new ArrayList<>(threadCount);
-		for (int t = 0; t < threadCount; t++) {
-			int offset = t * nrRandomizations / threadCount;
-			int count = (t + 1) * nrRandomizations / threadCount - offset;
-			future.add(es.submit(() -> {
-				final ShuffledView<T> shuffled = new ShuffledView<>(trimmedImage1, blockSize, seeds[offset]); // a new
-																												// one
-																												// per
-																												// thread
-																												// and
-																												// each
-																												// needs
-																												// its
-																												// own
-																												// seed
-				Img<T> buffer = Util.getSuitableImgFactory(shuffled, type1).create(shuffled);
-				for (int i = 0; i < count; i++) {
-					int index = offset + i;
-					if (index >= nrRandomizations)
-						break;
-					if (i > 0)
-						shuffled.shuffleBlocks(seeds[index]);
-					copy(shuffled, buffer);
-					sampleDistribution[index] = op.apply(buffer, trimmedImage2);
-				}
-			}));
-		}
 
-		// wait for threads to finish
-		try {
-			for (int t = 0; t < threadCount; t++) {
-				future.get(t).get();
+		List<Integer> params = IntStream.rangeClosed(0, numTasks - 1) //
+		 .boxed().collect(Collectors.toList());
+
+		Consumer<Integer> task = (t) -> {
+			int offset = t * nrRandomizations / numTasks;
+			int count = (t + 1) * nrRandomizations / numTasks - offset;
+			// a new one per thread and each needs its own seed
+			final ShuffledView<T> shuffled = new ShuffledView<>(trimmedImage1, blockSize, seeds[offset]);
+			Img<T> buffer = Util.getSuitableImgFactory(shuffled, type1).create(shuffled);
+			for (int i = 0; i < count; i++) {
+				int index = offset + i;
+				if (index >= nrRandomizations)
+					break;
+				if (i > 0)
+					shuffled.shuffleBlocks(seeds[index]);
+				copy(shuffled, buffer);
+				sampleDistribution[index] = op.apply(buffer, trimmedImage2);
 			}
-		} catch (final InterruptedException | ExecutionException exc) {
+		};
+
+		try {
+			Parallelization.getTaskExecutor().forEach(params, task);
```

Comment:
Didn't you get the `executor` above on line 92?

Reply:
Whoops, good catch! Fixed in f992fe5